pkg/util/kubernetes/kubelet/conversion.go

```diff
@@ -0,0 +1,348 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubelet && kubeapiserver
+
+package kubelet
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// ConvertKubeletPodToK8sPod converts a Pod to a Kubernetes Pod.
+// The Pod in this package is a simplification of the one in the Kubernetes
+// library, so the result will not contain all the fields. That's OK, because
+// this function is only called from the KSM check, so we only need to convert
+// the fields that are used by the check.
+func ConvertKubeletPodToK8sPod(pod *Pod) *corev1.Pod {
```

Comment:
do we plan to use the official corev1.Pod in the futur? or should we keep this `Pod` type for some reasons, Perf, backward compatibility?

Reply:
This pod is the one defined in our kubelet package. This is the one I used because it's more convenient given that it is what the pod watcher that we already have returns.

This is related to what I wanted to evaluate for future PRs. We have several pod representations and it might make sense to merge them, but it needs to be evaluated.